_knitr/A Shiny app for Hong Kong District Councillors_20200801.md

```diff
@@ -8,43 +8,160 @@ output:                    # DO NOT CHANGE
+- Contact email address
+- Contact number
+- Office address
+- Number of votes, and share of votes won in 2019
 
-## A solution for getting the Facebook feed
+A function that was extremely helpful for figuring out the URL of the District Councillors' individual official pages is the following. What this does is to run a Bing search on the https://www.districtcouncils.gov.hk website, and scrape from the search result any links which match what we want (based on what the URL string looks like). Although this doesn't always work, it helped us a long way with the 452 District Councillors.
+
+```
+scrape_dcs <- function(search_term){
+  
+  query_string <- paste("site: https://www.districtcouncils.gov.hk", search_term)
+  
+  squery <- URLencode(query_string)
+  
+  squeryfull <- paste0("https://www.bing.com/search?q=", squery)
+  
+  main_page <- xml2::read_html(squeryfull)
+  
+  temp <- html_nodes(main_page, '.b_title a') %>%
+    html_attr("href")
+  
+  temp[grepl("member_id=", temp)]
+}
+```
+One key thing to note is that all of the above data we compiled is available and accessible in the public domain, where we simply took an extra step to improve the accessibility.[^2] The Phase #2 data was used in the final app to provide more information to the user when a particular constituency or District Councillor is selected.
+
+[^2]: This is in compliance with the ICO's description of the 'public domain', i.e. that _information is only in the public domain if it is realistically accessible to a member of the general public at the time of the request. It must be available in practice, not just in theory_.
+
+### 📦 Creating a data package
+
+Our data R package, [**hkdatasets**](https://github.com/hong-Kong-Districts-Info/hkdatasets), is to some extent a spin-off of this project. We decided to migrate from Google Sheets to an R data package approach, for the following reasons: 
+
+- An R data package could allow us to provide more detailed documentation and tracking of how the data would change over time. If we choose to expand the dataset in the future, we can easily add this to the package release notes.
+
+- An R data package would fit well with our broader ambition to work on other Hong Kong themed, open-source projects. From sharing our project with friends, we were approached to help with another project to visualise Hong Kong traffic collisions data, where the repo is [here](https://github.com/Hong-Kong-Districts-Info/hktrafficcollisions). As part of this, we obtained this data via an FOI request on traffic collisions, where the data is also available through **hkdatasets**.
+
+- Make it easier for learners and students in the R community to practise with the datasets we've put together, without having to learn about the **googlesheets4** package. Our thinking is that this would benefit others as other data packages like **nycflights13** and **babynames** have benefitted us as we learned R.
```

Comment:
Are we using British English or American English? If the former, the below would be accurate.
```suggestion
- Make it easier for learners and students in the R community to practice with the datasets we've put together, without having to learn about the **googlesheets4** package. Our thinking is that this would benefit others as other data packages like **nycflights13** and **babynames** have benefitted us as we learned R.
```

Reply:
> Are we using British English or American English? If the former, the below would be accurate.

I thought with British English, _practise_ is a verb and _practice_ is a noun?